sympy/polys/matrices/sdm.py

```diff
@@ -970,3 +1017,22 @@ def sdm_particular_from_rref(A, ncols, pivots):
         if Ain is not None:
             P[j] = Ain / A[i][j]
     return P
+
+
+def sdm_applyfunc(M, shape, f, domain):
+    if not f(domain.zero):
+        return unop_dict(M, f)
+
+    nrows, ncols = shape
+    fzero = f(domain.zero)
+    Mf = {i: {j: fzero for j in range(ncols)} for i in range(nrows)}
+
+    for i, Mi in M.items():
+        for j, Mij in Mi.items():
+            fMij = f(Mij)
+            if fMij:
+                Mf[i][j] = fMij
+            else:
+                del Mf[i][j]
```

Comment:
This needs to check if row `i` becomes all zero in which case it should be removed. A test should be added for this case.